From b6935f58b302f3955b6622aeb056a2be7fe04fa8 Mon Sep 17 00:00:00 2001
From: Kareem Shaik <kareem.m.shaik@intel.com>
Date: Wed, 29 Nov 2017 15:38:20 +0530
Subject: [PATCH 0701/1132] ASoC: Intel: Skylake: Improve BXT-P machine driver
 selection based on configuration

Use configuration option to select the correct machine driver defined under
same machine id. Otherwise second machine driver is not loading without
a configuration check.

Change-Id: I419125f133f6b684916e81f26d0e5e685414a5af
Signed-off-by: Kareem Shaik <kareem.m.shaik@intel.com>
Reviewed-on:
Reviewed-by: Gogineni, GiribabuX <giribabux.gogineni@intel.com>
Reviewed-by: Kesapragada, Pardha Saradhi <pardha.saradhi.kesapragada@intel.com>
Reviewed-by: Kale, Sanyog R <sanyog.r.kale@intel.com>
Reviewed-by: Kp, Jeeja <jeeja.kp@intel.com>
Tested-by: Madiwalar, MadiwalappaX <madiwalappax.madiwalar@intel.com>
---
 sound/soc/intel/skylake/skl.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/intel/skylake/skl.c b/sound/soc/intel/skylake/skl.c
index 126b455..7b3faf8 100644
--- a/sound/soc/intel/skylake/skl.c
+++ b/sound/soc/intel/skylake/skl.c
@@ -1243,16 +1243,19 @@ IS_ENABLED(CONFIG_SND_SOC_INTEL_BXTP_IVI_GENERIC_MACH)
 		.machine_quirk = snd_soc_acpi_codec_list,
 		.quirk_data = &bxt_codecs,
 	},
+#if IS_ENABLED(CONFIG_SND_SOC_INTEL_BXT_TDF8532_MACH)
 	{
 		.id = "INT34C3",
 		.drv_name = "bxt_tdf8532",
 		.fw_filename = "intel/dsp_fw_bxtn.bin",
 	},
+#elif IS_ENABLED(CONFIG_SND_SOC_INTEL_BXT_ULL_MACH)
 	{
 		.id = "INT34C3",
 		.drv_name = "bxt_ivi_ull",
 		.fw_filename = "intel/dsp_fw_ull_bxtn.bin",
 	},
+#endif
 	{}
 };
 
-- 
2.7.4

